common/predictive-text/unit_tests/headless/default-word-breaker.js

```diff
@@ -68,7 +74,9 @@ describe('The default word breaker', function () {
   });
     
   it('returns empty string when called without word text', function() {
-    var model = new TrieModel(jsonFixture('tries/english-1000'));
+    var model = new TrieModel(jsonFixture('tries/english-1000'), {
+      wordBreaker: breakWords
+    });
 
     // Wordbreaking on a empty space => no word.
     context = { 
```

Comment:
I'm actually really confused by this test case.

How can the context have both text to the right and the left of the cursor, **and** be at the start of the buffer and at the end of the buffer?

What exactly is this test testing? What does it mean for the word breaker to be told to break words "without word text"?

I think what I'm getting at is, **what is the [contract](https://en.wikipedia.org/wiki/Design_by_contract) that a word breaker should follow**?

I'm assuming this is the code that we want:

```suggestion
    context = {
           left: 'The quick brown fox jumped ', startOfBuffer: false,
            right: 'over the lazy dog.', endOfBuffer: false
     };
```

And given this context where the cursor is after whitespace, it should return an empty string?

Reply:
This example is taken directly from worker-communication-protocol.md, under the "**Message: `currentword`**" section.

That said, examining index.d.ts for lexical-model-types:

```
  /**
   * Whether the insertion point is at the start of the buffer.
   */
  readonly startOfBuffer: boolean;

  /**
   * Whether the insertion point is at the end of the buffer.
   */
  readonly endOfBuffer: boolean;
```

It looks like there was a shift in the specification at some point; the example referenced above and the test case seem to instead use (for `startOfBuffer`) "Whether the text of `startOfBuffer` is at the start of the buffer."

One or the other needs to be updated accordingly, but that appears to be outside of this particular PR's scope.

Reply:
Also, yes, that should return empty string.  This is what's breaking MTNT 0.1.3 right now.